File: solutions/python/leap/1/leap.py
def leap_year(year: int) -> bool:
    divisible_by4 = year % 4 == 0
    divisible_by100 = year % 100 == 0
    divisible_by400 = year % 400 == 0

    return divisible_by400 or (divisible_by4 and not divisible_by100)

File: solutions/c/leap/1/leap.c
#include "leap.h"

bool leap_year(int year) {
  bool divisibleBy4 = year % 4 == 0;
  bool divisibleBy100 = year % 100 == 0;
  bool divisibleBy400 = year % 400 == 0;

  return divisibleBy400 || (divisibleBy4 && !divisibleBy100);
}
